server/src/dispatcher.erl

```diff
@@ -0,0 +1,145 @@
+%%%-------------------------------------------------------------------
+%% @doc Dispatcher API
+%% @end
+%%%-------------------------------------------------------------------
+-module(dispatcher).
+
+%% API
+-export([start_link/0]).
+%% gen_server callbacks
+-export([init/1, handle_call/3, handle_cast/2, handle_info/2, terminate/2,
+         code_change/3]).
+
+-define(SERVER, lyceum_server).
+
+-include("types.hrl").
+
+%%%===================================================================
+%%% API
+%%%===================================================================
+
+%%--------------------------------------------------------------------
+%% @doc
+%% Starts the server
+%%
+%% @spec start_link() -> {ok, Pid} | ignore | {error, Error}
+%% @end
+%%--------------------------------------------------------------------
+start_link() ->
+    gen_server:start_link({local, ?SERVER}, ?MODULE, [], []).
+
+%%%===================================================================
+%%% gen_server callbacks
+%%%===================================================================
+
+%%--------------------------------------------------------------------
+%% @private
+%% @doc
+%% Initializes the server
+%%
+%% @spec init(Args) -> {ok, State} |
+%%                     {ok, State, Timeout} |
+%%                     ignore |
+%%                     {stop, Reason}
+%% @end
+%%--------------------------------------------------------------------
+init([]) ->
+    % Setup a DB connection and bootstrap process state
+    {ok, Connection} = database:database_connect(),
+    Pid = self(),
+    io:format("[~p] Starting at ~p...~n", [?SERVER, Pid]),
+    State = #state{connection = Connection, pid = self()},
+    {ok, State}.
+
+%%--------------------------------------------------------------------
+%% @private
+%% @doc
+%% Handling call messages
+%%
+%% @spec handle_call(Request, From, State) ->
+%%                                   {reply, Reply, State} |
+%%                                   {reply, Reply, State, Timeout} |
+%%                                   {noreply, State} |
+%%                                   {noreply, State, Timeout} |
+%%                                   {stop, Reason, Reply, State} |
+%%                                   {stop, Reason, State}
+%% @end
+%%--------------------------------------------------------------------
+handle_call(_Request, _From, State) ->
+    Reply = ok,
+    {reply, Reply, State}.
```

Comment:
I do understand that `Reply = ok` is there for documentation purposes (instead of being inlined), but given the docs above, is this really needed? Can't we just return `{reply, ok, State}` without that much hit to readability?

Reply:
We can, but i'll let things sstay like this because we are not even using the `handle_call` ad `handle_cast` properly yet.